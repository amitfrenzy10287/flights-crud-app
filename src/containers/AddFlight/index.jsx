import React,{useState,useCallback} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import * as actions from "../../store/actions";
import {connect} from "react-redux";
import axios from "../../axios-flights";
import Alert from '@material-ui/lab/Alert';
import AlertTitle  from '@material-ui/lab/AlertTitle';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import CloseIcon from '@material-ui/icons/Close';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';

const useStyles = makeStyles((theme) => ({
    root: {
        '& .MuiTextField-root': {
            margin: theme.spacing(1),
            width: 200,
        },
    },
    containerWrap: {
        padding: theme.spacing(2),
        minHeight: 500,
        height: 500
    },
    typeContainer: {
        flexDirection:'row',
        padding: theme.spacing(2),
    }
}));

const AddFlight=(props)=> {
    const classes = useStyles();
    const [flightdata, setFlightdata] = useState({
       arrival: '',
       departure: '',
       arrivalTime: '07:30',
       departureTime: '15:30',
       class: 'business'
    });
    const [open, setOpen] = useState(true);
    const [isValidate, setValidate] = useState(true);
    const onInputChange = useCallback((e, type)=>{
        const data = {};
        data[type] = e.target.value;
        setFlightdata(prevState => ({
            ...prevState,
            ...data
        }));
    },[flightdata]);

    const saveFlights=useCallback(()=>{
        setFlightdata({
            arrival:'',
            departure:'',
            arrivalTime:'',
            departureTime:'',
            class:''
        });
        if(
            flightdata.arrival ===''
            || flightdata.departure ===''
            || flightdata.arrivalTime ===''
            || flightdata.departureTime ===''
            || flightdata.class ===''){
            setValidate(false);
        }else{
            setValidate(true);
            props.addFlights(flightdata);
        }

    },[flightdata]);
    const handleBack =()=>{
      props.history.push('/');
    };
    const handleTypeChange=useCallback(()=>{

    },[]);
    const { flightsRes } = props;
    return (
        <Grid lg={6} xl={6} mg={6} item justify="center">
            <Grid>
                { flightsRes && flightsRes ==='Created' && open === true &&
                <Alert
                    action={
                        <IconButton
                            aria-label="close"
                            color="inherit"
                            size="small"
                            onClick={() => {
                                setOpen(false);
                            }}
                        >
                            <CloseIcon fontSize="inherit" />
                        </IconButton>
                    }
                >
                    Flight added successfully!
                </Alert> }
            </Grid>
            <Paper className={classes.containerWrap}>
                {!isValidate && <Alert severity="error">
                    <AlertTitle>Error</AlertTitle>
                    All field required â€” <strong>Please fill all the detail's to continue!</strong>
                </Alert>}
                <Box p={2}>
                    <Typography variant="h5" color="primary" component="h5">
                         Add Flights
                    </Typography>
                </Box>
                <form className={classes.root} noValidate autoComplete="off">
                    <div>
                        <TextField
                            label="Arrival" type="text"
                            onChange={(e)=>onInputChange(e,'arrival')}
                            value={flightdata.arrival}
                            variant="filled"
                        />
                        <TextField
                            label="Departure" type="text"
                            onChange={(e)=>onInputChange(e,'departure')}
                            value={flightdata.departure}
                            variant="filled"
                        />
                    </div>
                    <div>
                        <TextField
                            label="Arrival Time"
                            onChange={(e)=>onInputChange(e,'arrivalTime')}
                            value={flightdata.arrivalTime}
                            variant="filled"
                            type="time"
                            InputLabelProps={{
                                shrink: true,
                            }}
                            inputProps={{
                                step: 300, // 5 min
                            }}
                        />
                        <TextField
                            label="Departure Time"
                            onChange={(e)=>onInputChange(e,'departureTime')}
                            value={flightdata.departureTime}
                            variant="filled"
                            type="time"
                            InputLabelProps={{
                                shrink: true,
                            }}
                            inputProps={{
                                step: 300, // 5 min
                            }}
                        />
                    </div>
                    <div>
                        <RadioGroup className={classes.typeContainer}
                            aria-label="type" name="type" value={flightdata.class}
                            onChange={(e)=>onInputChange(e,'class')}>
                            <FormControlLabel value="business" control={<Radio />} label="Business Class" />
                            <FormControlLabel value="cheap" control={<Radio />} label="Economy Class" />
                        </RadioGroup>
                    </div>
                </form>
                <Grid container justify="left" item>
                    <Box m={2}>
                        <Button onClick={saveFlights} variant="contained" color="primary">
                            Save & Continue
                        </Button>
                    </Box>
                    <Box m={2}>
                        <Button onClick={handleBack} variant="contained" >
                            Back
                        </Button>
                    </Box>
                </Grid>
            </Paper>
        </Grid>
    );
};
const mapStateToProps = state => {
    return {
        flightsRes: state.flights && state.flights.addFlight ? state.flights.addFlight : ''
    }
};

const mapDispatchToProps = dispatch => {
    return {
        addFlights: (data) => dispatch(actions.addFlights(data)),
    }
};
export default connect(mapStateToProps, mapDispatchToProps)( AddFlight, axios );