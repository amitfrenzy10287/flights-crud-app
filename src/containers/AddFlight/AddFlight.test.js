import React from 'react';
import AddFlight from './index';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

configure({adapter: new Adapter()});
const mockStore = configureMockStore();
const store = mockStore({});

describe("To test the title exist", () => {
    it("should render without throwing an error", () => {
        expect(
            shallow(
                <Provider store={store}>
                    <AddFlight />
                </Provider>
            ).exists(<h5>Add Flights</h5>)
        );
    });
});

describe('<AddFlight /> with no props', () => {
    const state = {
        addFlight:''
    };
    const container = shallow(
        <Provider store={store}>
            <AddFlight />
        </Provider>
    );
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('should have a btn component', ()=> {
        //There should be only one button
        expect(container.find('Button'));
    });

});