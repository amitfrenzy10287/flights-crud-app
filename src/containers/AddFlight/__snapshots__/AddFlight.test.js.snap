// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddFlight /> with no props should match the snapshot 1`] = `"<div class=\\"MuiGrid-root MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-lg-6 MuiGrid-grid-xl-6\\" mg=\\"6\\"><div class=\\"MuiGrid-root\\"></div><div class=\\"MuiPaper-root makeStyles-containerWrap-2 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiBox-root MuiBox-root-136\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-colorPrimary\\">Add Flights</h5></div><form class=\\"makeStyles-root-1\\" novalidate=\\"\\" autoComplete=\\"off\\"><div><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled\\" data-shrink=\\"false\\">Arrival</label><div class=\\"MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" value=\\"\\" class=\\"MuiInputBase-input MuiFilledInput-input\\"/></div></div><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled\\" data-shrink=\\"false\\">Departure</label><div class=\\"MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" value=\\"\\" class=\\"MuiInputBase-input MuiFilledInput-input\\"/></div></div></div><div><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled\\" data-shrink=\\"true\\">Arrival Time</label><div class=\\"MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl\\"><input type=\\"time\\" aria-invalid=\\"false\\" value=\\"07:30\\" step=\\"300\\" class=\\"MuiInputBase-input MuiFilledInput-input\\"/></div></div><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled\\" data-shrink=\\"true\\">Departure Time</label><div class=\\"MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl\\"><input type=\\"time\\" aria-invalid=\\"false\\" value=\\"15:30\\" step=\\"300\\" class=\\"MuiInputBase-input MuiFilledInput-input\\"/></div></div></div><div><div class=\\"MuiFormGroup-root makeStyles-typeContainer-3\\" role=\\"radiogroup\\" aria-label=\\"type\\"><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-239 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-240 Mui-checked MuiIconButton-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"radio\\" checked=\\"\\" class=\\"PrivateSwitchBase-input-242\\" name=\\"type\\" value=\\"business\\"/><div class=\\"PrivateRadioButtonIcon-root-255 PrivateRadioButtonIcon-checked-257\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg><svg class=\\"MuiSvgIcon-root PrivateRadioButtonIcon-layer-256\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z\\"></path></svg></div></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">Business Class</span></label><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-239 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input type=\\"radio\\" class=\\"PrivateSwitchBase-input-242\\" name=\\"type\\" value=\\"cheap\\"/><div class=\\"PrivateRadioButtonIcon-root-255\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg><svg class=\\"MuiSvgIcon-root PrivateRadioButtonIcon-layer-256\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z\\"></path></svg></div></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">Economy Class</span></label></div></div></form><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item\\"><div class=\\"MuiBox-root MuiBox-root-267\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Save &amp; Continue</span></button></div><div class=\\"MuiBox-root MuiBox-root-297\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Back</span></button></div></div></div></div>"`;
